File: server/src/main/kotlin/com/openmpy/server/chat/dto/response/ChatRoomGetResponse.kt
package com.openmpy.server.chat.dto.response

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime

data class ChatRoomGetResponse(

    val chatRoomId: Long,
    val thumbnail: String?,
    val nickname: String,
    val lastMessage: String,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS")
    val lastMessageAt: LocalDateTime,
)

File: app/app/Model/Chat/Response/ChatRoomGetResponse.swift
struct ChatRoomGetResponse: Codable, Identifiable {

    let chatRoomId: Int64
    let memberId: Int64
    let thumbnail: String?
    let nickname: String
    var lastMessage: String
    var lastMessageAt: String
    var unreadCount: Int64

    var id: Int64 { chatRoomId }
}
